### PR TITLE
Add setting to indicate a flag can be set multiple times

### LINKED ISSUE #111: Add setting to indicate a flag can be set multiple times
[Issuehunt badges](https://issuehunt.io/r/sindresorhus/meow/issues/111)

Currently, if you have:

```js
foo: {
	type: 'string'
}
```

And on the command-line, you do:

```
$ cli --foo=a
```

You get `'a'` back, but if you do:

```
$ cli --foo=a --foo=b
```

You get and array back `['a', 'b']`.

This is surprising and inconsistent.

Instead, we should allow only one value by default, and add a setting to the flags config to allow multiple for a flag, in which case, it will always return an array, even if there's only one flag.

I imagine it could be something like this:

```js
{
	flags: {
		foo: {
			type: 'string',
			multiple: true
		}
	}
}
```

> There is a $40.00 open bounty on this issue. Add more on [Issuehunt](https://issuehunt.io/r/sindresorhus/meow/issues/111).
> - Checkout the [Issuehunt explorer](https://issuehunt.io/r/sindresorhus/meow/) to discover more funded issues.
> - Need some help from other developers? [Add your repositories](https://issuehunt.io/r/new) on Issuehunt to raise funds.

### PR DESCRIPTION
**Pull request by ankeetmaini**
Fixes #111 

Taken inspiration and some test cases from #112

### CODE REVIEW

**Comment by ankeetmaini** [comment]
ping @sindresorhus :)

**Comment by sindresorhus** [comment]
You need to update index.d.ts too.

File: readme.md [hunk 1]
```diff
@@ -101,6 +101,7 @@ The key is the flag name and the value is an object with any of:
 - `type`: Type of value. (Possible values: `string` `boolean`)
 - `alias`: Usually used to define a short flag alias.
 - `default`: Default value when the flag is not specified.
+- `multiple`: To indicate that a flag can be passed multiple times, value will be an array. Default value is `false`
```
Comment (sindresorhus) [inline readme.md]:
```suggestion
- `multiple`: Indicates that a flag can be passed multiple times. The flag value will be an array. (Default: `false`)
```

File: test.js [hunk 2]
```diff
@@ -240,3 +240,67 @@ test('disable autoVersion/autoHelp if `cli.input.length > 0`', t => {
...
+	t.deepEqual(meow({
+		flags,
+		argv: [
+			'--foo=a',
+			'--foo=b',
+			'--foo=c',
+			'--foo=d'
+		]
+	}).flags, {foo: ['a', 'b', 'c', 'd']});
+});
```
Comment (sindresorhus) [inline test.js]: Can you add tests to ensure it works correctly with `flags.type == 'boolean'` and `flags.type == 'number'`?

File: test.js [hunk 3]
```diff
@@ -240,3 +240,67 @@ test('disable autoVersion/autoHelp if `cli.input.length > 0`', t => {
...
+
+test('combination of multi flag and single', t => {
+	const cli = meow({
+		argv: ['--foo=1', '--foo=2', '--bar', '1', '--foo', '3', '--bar=5', '-f', '4'],
+		flags: {
+			foo: {multiple: true, alias: 'f', type: 'string'},
+			bar: {type: 'string'}
+		}
+	});
+	t.deepEqual(cli.flags.foo, [1, 2, 3, 4]);
```
Comment (sindresorhus) [inline test.js]: Shouldn't the elements in the array be strings?

**Comment by sindresorhus** [comment]
Bump :)

**Pull request closed without merging**